roles/harden/tasks/main.yaml

```diff
@@ -21,12 +21,28 @@
     create_home: true
 
 
-- name: Set authorized key for remote admin
-  ansible.posix.authorized_key:
-    user: "{{ admin_username }}"
-    state: present
-    key: "{{ lookup('file', admin_ssh_pubkey) }}"
-  register: admin_added
+- name: Create .ssh folder for admin user
+  ansible.builtin.file:
+    path: "{{ homedir }}/.ssh/"
+    state: directory
+    mode: "0700"
+    owner: "{{ admin_username }}"
+    group: users
+
+
+- name: Copy authorized keys from root dir to admin dir
```

Comment:
This means that the keys won't be updated if they change as far as I understand, maybe we should add root's authorized keys to ssh config using `AuthorizedKeysFile` instead?

Reply:
I am not sure I follow.  When would the root's keys change?  After this playbook, there shouldn't be updates to root's ssh keys, as you cannot log into root at all.  This change takes all the existing keys in the authorized_keys file and copies them to the admin dir, and then if you wanted to add people, you'd add keys to the admin's dir.  But I think I'm misunderstanding your comment maybe.

Reply:
I'm not sure if this is exactly what Filip was pointing out, but I do think this step breaks the ansible rule of idempotency. I don't think this role is idempotent in this scenario:
1. Create a digital ocean droplet
2. run the new-server-playbook
3. manually add a key /home/admin/.ssh/authorized_keys
4. run the new-server-playbook again
In this case the manually added key will be wiped out.

One way to solve this would be to have an authorized_keys file in the ansible repo that we just copy to all new servers. Then if you want to add someone you add them to the authorized keys file in this repo and re-run the new-server-playbook. The disadvantage is that if you want certain people to only have access to certain servers it could get hairy.

Another solution (which solves other problems too like centralized logging and preventing concurrently running playbooks) is to set up a control server and only allow the control server to access the others via SSH. Then you just add team member's keys to the control server. 

Do either of these sound like a good plan?